dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLlvm.java

```diff
@@ -309,20 +309,28 @@ private List<Metadata> parseMetadataAttachment(List<MetadataAttachmentContext> m
         final List<Metadata> metadata = new ArrayList<>();
         //FIXME: This code only looks for DILocation metadata,
         // and it only extracts the information needed to construct SourceLocation metadata
-        for (MetadataAttachmentContext metadataCtx:  metadataAttachmentContexts) {
+        for (MetadataAttachmentContext metadataCtx : metadataAttachmentContexts) {
             MdNode mdNode = (MdNode) metadataCtx.accept(this);
             assert mdNode instanceof MdReference;
             mdNode = metadataSymbolTable.get(((MdReference) mdNode).mdName());
 
-            if (mdNode instanceof SpecialMdTupleNode diLocationNode && diLocationNode.nodeType() == SpecialMdTupleNode.Type.DILocation) {
+            if (mdNode instanceof SpecialMdTupleNode diLocationNode
+                    && diLocationNode.nodeType() == SpecialMdTupleNode.Type.DILocation) {
                 SpecialMdTupleNode scope = diLocationNode;
                 while (scope.getField("scope").isPresent()) {
-                    scope = (SpecialMdTupleNode) metadataSymbolTable.get(scope.<MdReference>getField("scope").orElseThrow().mdName());
+                    scope = (SpecialMdTupleNode) metadataSymbolTable
+                            .get(scope.<MdReference>getField("scope").get().mdName());
                 }
                 assert scope.nodeType() == SpecialMdTupleNode.Type.DIFile;
-                final String filename = scope.<MdGenericValue<String>>getField("filename").orElseThrow().value();
-                final String directory = scope.<MdGenericValue<String>>getField("directory").orElseThrow().value();
-                final int lineNumber = diLocationNode.<MdGenericValue<BigInteger>>getField("line").orElseThrow().value().intValue();
+                final String filename = scope.getField("filename").isPresent()
+                        ? scope.<MdGenericValue<String>>getField("filename").get().value()
+                        : "";
+                final String directory = scope.getField("directory").isPresent()
+                        ? scope.<MdGenericValue<String>>getField("directory").get().value()
+                        : "";
+                final int lineNumber = scope.getField("line").isPresent()
+                        ? diLocationNode.<MdGenericValue<BigInteger>>getField("line").get().value().intValue()
+                        : 0; // Assuming line 0 when value not present
```

Comment:
You can just use `map.(n -> n.value).orElse(defaultValue)` (*) in place of the original `orElseThrow()`.
What precisely is happening though? Do we get just some of the fields rather than all?
I thought those are special metadata nodes where all fields are required, otherwise the IR is ill-formed (maybe I'm misremembering though).

(*) written from memory, so might need some adaption :P):

Reply:
Line is optional, see LLVM parser: https://github.com/llvm/llvm-project/blob/main/llvm/lib/AsmParser/LLParser.cpp#L5049

```c++
/// parseDILocationFields:
///   ::= !DILocation(line: 43, column: 8, scope: !5, inlinedAt: !6,
///   isImplicitCode: true)
bool LLParser::parseDILocation(MDNode *&Result, bool IsDistinct) {
#define VISIT_MD_FIELDS(OPTIONAL, REQUIRED)                                    \
  OPTIONAL(line, LineField, );                                                 \
  OPTIONAL(column, ColumnField, );                                             \
  REQUIRED(scope, MDField, (/* AllowNull */ false));                           \
  OPTIONAL(inlinedAt, MDField, );                                              \
  OPTIONAL(isImplicitCode, MDBoolField, (false));
  PARSE_MD_FIELDS();
#undef VISIT_MD_FIELDS
```